src/Translator/assets/src/formatters/formatter.ts

```diff
@@ -0,0 +1,236 @@
+import {strtr} from '../utils';
+
+/**
+ * This code is adapted from the Symfony Translator Trait (6.2)
+ * @see https://github.com/symfony/symfony/blob/015d5015e353ee5448bf7c350de0db4a03f8e13a/src/Symfony/Contracts/Translation/TranslatorTrait.php
+ */
+
+/**
+ * Translates the given message.
+ *
+ * When a number is provided as a parameter named "%count%", the message is parsed for plural
+ * forms and a translation is chosen according to this number using the following rules:
+ *
+ * Given a message with different plural translations separated by a
+ * pipe (|), this method returns the correct portion of the message based
+ * on the given number, locale and the pluralization rules in the message
+ * itself.
+ *
+ * The message supports two different types of pluralization rules:
+ *
+ * interval: {0} There are no apples|{1} There is one apple|]1,Inf] There are %count% apples
+ * indexed:  There is one apple|There are %count% apples
+ *
+ * The indexed solution can also contain labels (e.g. one: There is one apple).
+ * This is purely for making the translations more clear - it does not
+ * affect the functionality.
+ *
+ * The two methods can also be mixed:
+ *     {0} There are no apples|one: There is one apple|more: There are %count% apples
+ *
+ * An interval can represent a finite set of numbers:
+ *  {1,2,3,4}
+ *
+ * An interval can represent numbers between two numbers:
+ *  [1, +Inf]
+ *  ]-1,2[
+ *
+ * The left delimiter can be [ (inclusive) or ] (exclusive).
+ * The right delimiter can be [ (exclusive) or ] (inclusive).
+ * Beside numbers, you can use -Inf and +Inf for the infinite.
+ *
+ * @see https://en.wikipedia.org/wiki/ISO_31-11
+ *
+ * @param id         The message id
+ * @param parameters An array of parameters for the message
+ * @param locale     The locale
+ */
+export function format(id: string, parameters: Record<string, string | number> = {}, locale: string): string {
```

Comment:
From my initial understanding, this logic is basically private (it's used by your trans function but it's not meant to be used outside of the module)?

If that's true, maybe we can add a `@internal` annotation or something similar to discourage the direct use of this function outside of the package, to avoid BC breaks.

Reply:
Yes! This is something we can add or internal files

Reply:
Looks like the equivalent tag for `@internal` is `@private`, see https://jsdoc.app/tags-private.html